sheetScript.gs

```diff
@@ -63,94 +70,101 @@ function updatePlaylists(sheet) {
-         playlistIds.push(channel);
-      else if (!(channel.substring(0,2) == "UC" && channel.length > 10)) // Check if it is not a channel ID (therefore a username). MaybeTODO: do a better validation, since might interpret a channel with a name "UC..." as a channel ID
-      {
-        try {
-          var user = YouTube.Channels.list('id', {forUsername: channel, maxResults: 1});
-          if (!user || !user.items) addError("Cannot query for user " + channel)
-          else if (user.items.length === 0) addError("No user with name " + channel)
-          else if (user.items.length !== 1) addError("Multiple users with name " + channel)
-          else if (!user.items[0].id) addError("Cannot get id from user " + channel)
-          else channelIds.push(user.items[0].id);
-        } catch (e) {
-          addError("Cannot search for channel with name "+channel+", ERROR: " + "Message: [" + e.message + "] Details: " + JSON.stringify(e.details));
-          continue;
-        }
-      }
-      else
-        channelIds.push(channel);
-    }
-    
-    var lastTimestamp = sheet.getRange(iRow + 1, reservedTimestampColumn + 1).getValue();
+    var lastTimestamp = data[iRow][reservedColumnTimestamp];
     if (!lastTimestamp) {
       var date = new Date();
       date.setHours(date.getHours() - 24); // Subscriptions added starting with the last day
       var isodate = date.toISOString();
-      sheet.getRange(iRow + 1, reservedTimestampColumn + 1).setValue(isodate);
+      sheet.getRange(iRow + 1, reservedColumnTimestamp + 1).setValue(isodate);
       lastTimestamp = isodate;
     }
-
-    /// ...get videos from the channels...
-    var newVideoIds = [];
-    for (var i = 0; i < channelIds.length; i++) {
-      var videoIds = getVideoIdsWithLessQueries(channelIds[i], lastTimestamp)
-      if (!videoIds || typeof(videoIds) !== "object") addError("Failed to get videos with channel id "+channelIds[i])
-      else if (debugFlag_logWhenNoNewVideosFound && videoIds.length === 0) {
-        Logger.log("Channel with id "+channelIds[i]+" has no new videos")
-      } else {
-        [].push.apply(newVideoIds, videoIds);
-      }
-    }
-    for (var i = 0; i < playlistIds.length; i++) {
-      var videoIds = getPlaylistVideoIds(playlistIds[i], lastTimestamp)
-      if (!videoIds || typeof(videoIds) !== "object") addError("Failed to get videos with playlist id "+playlistIds[i])
-      else if (debugFlag_logWhenNoNewVideosFound && videoIds.length === 0) {
-        Logger.log("Playlist with id "+playlistIds[i]+" has no new videos")
-      } else {
-        [].push.apply(newVideoIds, videoIds);
+  
+  // Check if it's time to update already
+  var freqDate = new Date(lastTimestamp);
+  var dateDiff = Date.now() - freqDate;
+  var nextTime = data[iRow][reservedColumnFrequency]  * MILLIS_PER_HOUR;
+  if (nextTime && dateDiff <= nextTime) {
+    Logger.log("Skipped: Not time yet");
+  } else {
```

Comment:
These lines should be 6-space indented for consistency with code above and below